File: code/community/1304606470934892553/zero-duration-live-transcriptions.js
require('dotenv').config();
const { Deepgram } = require('@deepgram/sdk');

const deepgramApiKey = process.env.DEEPGRAM_API_KEY;

const deepgram = new Deepgram(deepgramApiKey);

const options = {
    punctuate: true,
    encoding: 'mulaw',
    channels: 2,
    sample_rate: 8000,
    multichannel: true // Ensure multichannel is set to true for multiple channels
};

deepgram.transcription.preRecorded(options)
    .then((response) => {
        console.log('Live transcription started successfully:', response);
    })
    .catch((error) => {
        console.error('Error starting live transcription:', error);
    });

// For more about multichannel setup, refer to: https://developers.deepgram.com/docs/multichannel

File: code/community/1304606470934892553/zero-duration-live-transcriptions.rs
use std::env;
use deepgram::{Deepgram, LiveTranscriptionOptions};
use tokio;

#[tokio::main]
async fn main() {
    let api_key = env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY must be set");

    let options = LiveTranscriptionOptions {
        punctuate: true,
        encoding: "mulaw".into(),
        channels: Some(2),
        sample_rate: Some(8000),
        multichannel: Some(true) // Ensure multichannel is set to true for multiple channels
    };

    let deepgram = Deepgram::new(&api_key);

    match deepgram.start_live_transcription(options).await {
        Ok(response) => println!("Live transcription started successfully: {:?}", response),
        Err(e) => eprintln!("Error starting live transcription: {:?}", e),
    }
}

// For more about multichannel setup, refer to: https://developers.deepgram.com/docs/multichannel